### ISSUE TITLE
Display shape name and library name of shape on canvas

### ISSUE THREAD

**Issue by mararad** [jgraph/drawio#913]
As a DrawIO user, I would find it useful if I could:

* See the shape name when selecting it on the DrawIO page canvas.
* See the shape library name from where the shape copied to the canvas.

My request is based on my use of the AWS shape libraries. I want to use the correct shape so others can read my diagrams and design clearly and without confusion. However, on a diagram with many AWS shapes, I constantly have to refer back to the library when I forget which shape matches to which AWS resource.

Perhaps as the Style tab in DrawIO has a table of Property/Value pairs, the Arrange tab could have a Shape Name Library Name that shows the name when the shape is selected. I would expect the names to be read-only.

The shapes and libraries already have names. They just need to be visible once selected on the canvas.

Chat user 27068446925

**Comment by webketje** [comment]
FWIW you can do this with a manual workaround:

1. Add the shape from the library, and briefly memorize its name.
2. Select the shape > Edit Data > Add Property with name "tooltip", and write its name as value.

You should now be able to see the shape name when hovering with your cursor over it. This tooltip will also be visible in HMTL exports.

![](https://i.imgur.com/2T9U7tb.png)

**Comment by rhoering** [comment]
I would like to ask that this feature can be revisited. I searched issues and discussions but I did not see if the feature was implemented under any of the other issue threads.

I am happy to create a new feature request if the following is different than the request above.

Feature: Have the name of the shape as seen when hovering over the shape in the library be a default tooltip when the shape is placed on the canvas. 

Option: Add the library shape name as a visible and togglable Property of the on-canvas shape's as seen on the Style tab of the Format pane.

On a related note, if the library shape name could also be added to the on-canvas shape as an automatic tag that would also be a wonderful feature.

Thank you in advance for your consideration and for this wonder program.

Kind regards,

Ron
